lib/laforge/activerecord.rb

```diff
@@ -59,13 +59,24 @@ def forge_diff(**forge_options)
     # Optionally pass `sources` to limit data entries used in the calculation to only those from the given sources
     def forge_attributes(attributes: nil, sources: nil)
       forged_attrs = {}
+
       filter_loaded_data_entries(attributes: attributes, sources: sources, present: true).sort_by(&:priority_with_fallback).reverse.uniq(&:attribute_name).each do |data_entry|
-        forged_attrs[data_entry.attribute_name] = data_entry.value
+        value = data_entry.marked_for_destruction? ? nil : data_entry.value
+        forged_attrs[data_entry.attribute_name] = value
       end
 
       return forged_attrs
     end
 
+    # Mark several data entries for destruction
+    # Optionally pass `attributes` to limit data entries used in the calculation to only those for the given attributes
+    # Optionally pass `sources` to limit data entries used in the calculation to only those from the given sources
+    def remove_data_entries(attribute_names: nil, sources: nil)
```

Comment:
Only suggestion would be to consider the name of this method. `remove_data_entries` sounds like it happens immediately but requires a save. Compare this to `record_data_entries` which applies changes immediately. In fact, it even deletes data entries immediately.

I'd suggest having both methods make modifications that require a save of the record. You can then add ! versions of each if desired, which would match up with the `forge` vs `forge!` naming pattern.